### PR TITLE
fix: It is still possible to mint two identic images

### LINKED ISSUE #10312: It is still possible to mint two identic images
i remember that in the past we added a piece of code that would refresh the drop page after a mint was successful in order to reset the page states, but it was done as a temporary solution. 

it would be interesting to refactor/change the code in a way that this could be handled better.

in the flow below you can observe that i can still mint identical images by minting one after another(using auto-teleport instead of normal mint):

[twosameimages.webm](https://github.com/kodadot/nft-gallery/assets/36627808/8cd0a31f-335e-4659-9f2b-89de0655a093)

https://canary.kodadot.xyz/ahk/collection/379

image

### LINKED ISSUE #10335: Stage is changed before the modal animation is completed
small ui issue, stage is changed before the modal animation is completed

 can we make `NeoModal` emit an event or any other solution , so we don't have to [this](https://github.com/Jarsen136/nft-gallery/blob/fd3e24a9b0110919a6081f191e136f4931db7a0c/components/collection/drop/modal/PaidMint.vue#L198) all the time 

https://github.com/kodadot/nft-gallery/assets/44554284/9e0520dc-365a-4654-b30f-b2c0201685ac

_Originally posted by @hassnian in https://github.com/kodadot/nft-gallery/issues/10326#issuecomment-2124292101_

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

> using auto-teleport instead of normal mint

@prury you shouldn't have to use auto-teleport to replicate this, the page refresh is done after you close the success modal, opening the listing modal and then closing it allows you to skip it.

> it would be interesting to refactor/change the code in a way that this could be handled better.

#10148 fixes that

- [x] Closes #10312
- [x] Closes #10335

## Needs QA check

- @kodadot/qa-guild please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

https://github.com/kodadot/nft-gallery/assets/44554284/aea84e51-5375-42e4-9076-6255b1377c63

### CODE REVIEW

**Comment by prury** [comment]
> 
> @prury you shouldn't have to use auto-teleport to replicate this, the page refresh is done after you close the success modal, opening the listing modal and then closing it allows you to skip it.

i see, ty.

but i think the problem still occurs, i was able to close the listing modal in this PR as well without page refresh when closing.

**Comment by hassnian** [comment]
@prury  

## issue 1: page state not resetting

I have deleted the page refresh , let's see if the page app resets properly after a mint

pls test back to back mints using the normal  flow and also with auto-teleport and combine 

 ## issue 2: user's mint count not being updated after mint

now after a mint it moves to next range, this generates a new hash 

as you can see now you can mint multiple times and it's not giving you the same nft 

https://github.com/kodadot/nft-gallery/assets/44554284/7ba7faef-3f24-47a6-a731-01379b399f6d

## issue 3: modal state is being reset before the animation ends

I've also fixed https://github.com/kodadot/nft-gallery/issues/10335

https://github.com/kodadot/nft-gallery/assets/44554284/3cc053c5-69c4-4713-bfde-15ab869651d7

**Comment by prury** [comment]
> issue 1: page state not resetting after a mint without a page reset
> 
> I have deleted the page refresh , let's see if the app resets properly after a mint
> 
> pls test back to back mints using the normal flow and also with auto-teleport and combine
> ## issue 2: user's mint count not being updated after mint
> 
> now after a mint it moves to next entropy range, this generates a new hash , (00:45)
> 

nice, i could observe that the count was properly updated and a new hash is being generated as well in both auto teleport and normal mint without having to do a page refresh. 

> ## issue 3: modal state is being reset before the animation ends
> 
> #10335

nice, thank you for solving this one as well

everything that i tested so far worked, but i want to revisit this PR and do more tests, will wait for dev reviews.

File: components/collection/drop/PaidGenerative.vue [hunk 1]
```diff
@@ -116,7 +116,7 @@ const openMintModal = () => {
 
 const handleMintModalClose = () => {
   closeMintModal()
-  clearMassMint()
+  setTimeout(clearMassMint, NEO_MODAL_ANIMATION_DURATION)
 }
```
[the remark this answers is not in the document]
Comment (hassnian) [reply components/collection/drop/PaidGenerative.vue:120]: here we are not listening for `isModalActive`, this is reacting to the @close event , when going through the auto-teleport flow we set `isModalActive` to `false`, but in that case we don't want to clear

**Review by Jarsen136 — APPROVED** [review]

**Review by preschian — APPROVED** [review]
since we are changing how the user [get hash](https://github.com/kodadot/ops-internal/issues/1659), it solve this issue also

**Comment by prury** [comment]
> since we are changing how the user [get hash](https://github.com/kodadot/ops-internal/issues/1659), it solve this issue also

do you think we should close this PR then?

**Comment by preschian** [comment]
> do you think we should close this PR then?

let's merge this one

**Pull request merged**